content/post/2021-06-28-draw-me-a-project/index.md

```diff
@@ -0,0 +1,267 @@
+
+{{< figure src="https://raw.githubusercontent.com/MonkmanMH/EIKIFJB/main/EIKIFJB_sigmar_hex.png" alt="hex logo de 'Everything I know is from Jenny Bryan' avec un ordinateur portable en feu" width=350 link="https://github.com/MonkmanMH/EIKIFJB" >}}
+
+There's even a [logo](https://github.com/MonkmanMH/EIKIFJB) created by [Martin Monkman](https://github.com/MonkmanMH) for all of us agreeing with that statement! It features the letters EIKIFJB for "Everything I know is from Jenny Bryan" as well as a laptop on fire.
+
+Why a laptop on fire?!
+
+{{< tweet 940021008764846080 >}}
+
+That's a reference to a talk Jenny Bryan gave years ago, to which she said she'd put your computer on fire if you used `setwd("C:\Users\jenny\path\that\only\I\have")` or `rm(list = ls())` at the beginning of your scripts. This goes against the notion of a ["project-oriented workflow"](https://www.tidyverse.org/blog/2017/12/workflow-vs-script/). I do not need to repeat the wisdom of her blog post in mine, so will only use cliff notes:
+
+-   When reading in data etc. use paths relative to the root of your project, possibly using the [here package by Kirill Müller](https://github.com/jennybc/here_here).
+-   Re-start R often, do not e.g. load packages in your [.Rprofile](https://rstats.wtf/r-startup.html#rprofile), [`usethis::use_blank_slate()`](https://usethis.r-lib.org/reference/use_blank_slate.html).
+
+Really, go and read her post if you haven't yet!
+
+In my talk I'll show pictures of the Kei-tora Gardening Contest where participants create beautiful gardens in little trucks, which I find is a good image of a project that should be independent (you can move your truck, you should be able to move your project too).
+
+{{< tweet 1403009688392781824 >}}
+
+As the first part of my talk really is a collection of awesome content by Jenny Bryan, I will also show her rules for [naming files](http://www2.stat.duke.edu/~rcs46/lectures_2015/01-markdown-git/slides/naming-slides/naming-slides.pdf). What Shakespeare said about [roses](https://en.wikipedia.org/wiki/A_rose_by_any_other_name_would_smell_as_sweet) isn't true when programming. 😅
+
+Then I'll "introduce" version control. I'll briefly mention some basic git commands, and most importantly, how to run them! Here are my preferences:
+
+-   Using usethis ([`usethis::use_git()`](https://usethis.r-lib.org/reference/use_git.html), [`usethis::use_github()`](https://usethis.r-lib.org/reference/use_github.html), etc.) and gert ([`gert::git_push()`](https://docs.ropensci.org/gert/reference/git_fetch.html)) so that I don't need to leave R.
+
+-   Using RStudio git pane.
+
+-   Using the terminal for commands copy-pasted from somewhere on the web (and for the ones I now know by heart!).
+
+-   Using a git interface like GitKraken for more complicated stuff... which I haven't really ever fully explored, actually.
+
+I'll share links to my favorite git resources that I send to anyone who asks (or doesn't ask):
```

Comment:
Post is in english and I don't know if you shared more in the slides, but there is a french great resource from Thinkr: https://thinkr.fr/travailler-avec-git-via-rstudio-et-versionner-son-code/

Just in case you want other source than Jenny's 😉

Reply:
Great resource!